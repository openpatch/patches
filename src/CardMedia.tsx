/** @jsx jsx */
import { ReactNode, forwardRef, Ref, CSSProperties } from "react";
import { jsx, css } from "@emotion/react";
import { Properties } from "csstype";

export type CardMediaProps = {
  height?: Properties["height"];
  image?: string;
  title?: string;
  ref?: Ref<HTMLDivElement>;
};

export const CardMedia = forwardRef(
  ({ height, image, title }: CardMediaProps, ref: Ref<HTMLDivElement>) => {
    return (
      <div
        ref={ref}
        title={title}
        css={css`
          display: block;
          background-size: cover;
          background-repeat: no-repeat;
          background-position: center;
          background-image: url(${image});
          height: ${height};
        `}
      ></div>
    );
  }
);
