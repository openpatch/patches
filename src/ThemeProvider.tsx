import React, { ReactElement } from "react";
import { ThemeProvider as EThemeProvider } from "emotion-theming";
import { Theme, base } from "./themes";
import { GlobalCssTheme } from "./themes/utils";

interface ThemeProviderProps {
  theme?: Theme;
  children: ReactElement;
}

const ThemeProvider = ({ theme = base, children }: ThemeProviderProps) => {
  return (
    <EThemeProvider theme={theme}>
      <GlobalCssTheme theme={theme} />
      {children}
    </EThemeProvider>
  );
};

export default ThemeProvider;
