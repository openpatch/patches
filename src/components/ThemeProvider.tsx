import React, { ReactElement } from "react";
import {
  ThemeProvider as SCThemeProvider,
  DefaultTheme,
} from "styled-components";
import defaultTheme from "../shared/theme";

interface ThemeProviderProps {
  theme?: DefaultTheme;
  children: ReactElement;
}

const ThemeProvider = ({
  theme = defaultTheme,
  children,
}: ThemeProviderProps) => {
  return <SCThemeProvider theme={theme}>{children}</SCThemeProvider>;
};

export default ThemeProvider;
