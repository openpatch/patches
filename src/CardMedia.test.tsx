import React from "react";
import { render } from "@testing-library/react";
import { CardMedia, CardMediaProps } from "./CardMedia";
import { ThemeProvider } from "./ThemeProvider";

const renderWithTheme = (props?: CardMediaProps) =>
  render(
    <ThemeProvider>
      <CardMedia {...props} />
    </ThemeProvider>
  );

test("should render", () => {
  renderWithTheme();
});

test("should match snapshot", () => {
  const { container } = renderWithTheme();
  expect(container).toMatchSnapshot();
});
