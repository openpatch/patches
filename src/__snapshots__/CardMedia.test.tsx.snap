// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.emotion-0 {
  display: block;
  -webkit-background-size: cover;
  background-size: cover;
  background-repeat: no-repeat;
  -webkit-background-position: center;
  background-position: center;
  background-image: url();
  height: ;
}

<div>
  <div
    class="emotion-0"
  />
</div>
`;
